### DISCUSSION TITLE
What should I use as hyperparameters for the DefaultNS ?

### DISCUSSION THREAD

**Discussion by renecotyfanboy** [Joshuaalbert/jaxns#176] [Q&A] [answered]
Hi everyone! 
First, many thanks to Joshua for bringing us this awesome jax package for nested sampling, it is much appreciated. 
I am currently learning to use jaxns and nested sampling in general, but I struggle to find a good parametrization for the sampler. Comparing my runs with NUTS shows that I can't catch the desired correlations and values in my model, as shown below. For information, I am using the contributed code to numpyro to directly use their interface, as my models are already defined with this library.

![image](https://github.com/Joshuaalbert/jaxns/assets/49200287/15bb960f-beb9-499c-8da8-9f67c8aa388c)

I tried to increase the number of livepoints to 1000, and enabled "difficult_model" and "parameter_estimation", but this does not converge as intended. What would be your take on this ? Diagnostics below 

![image](https://github.com/Joshuaalbert/jaxns/assets/49200287/81fd73e0-d9dc-41d2-94ae-848334ef69d9)

Also if you have any kind of practical resources explaining the choice and meaning of each hyperparameter in addition to the existing papers, I would be glad to read them!

**Comment by Joshuaalbert** [comment]
Can you share a gist of the numpyro model? The posterior looks quite innocent such that default parameters should solve this no problem. Also, what version of jaxns are you using? `pip freeze | grep jaxns`

**Reply by renecotyfanboy** [reply]
Hi, thank you very much for your help. For you to know, I use a package I am developing to define the numpyro model so you might need to install it with `pip install jaxspec`. If you'd like, I might rewrite it in pure numpyro, but it will take me some time to do so, in the meantime, here is a [gist using `jaxspec`](https://gist.github.com/renecotyfanboy/3069171216a34ced78e9b67a84f4aa37). I am using `jaxns` 2.5.1 in my environment. 

Maybe this is due to the wrapping with numpyro ? I have no idea tbh

**Reply by renecotyfanboy** [reply]
Some news about this, I tested this same problem using a package implementing `ultranest` sampler, and got similar results to what is obtained from NUTS using the default parameters.

![comparison(1)](https://github.com/Joshuaalbert/jaxns/assets/49200287/e365bd63-d7e2-4ab0-b6b6-54dc67fc8572)

 I also tested the [Gaussian Shell](https://num.pyro.ai/en/stable/examples/gaussian_shells.html) example from `numpyro`+ `jaxns` to check if the contributed code is in default, but it seems to work properly too.

![Capture d’écran 2024-07-02 à 14 25 17](https://github.com/Joshuaalbert/jaxns/assets/49200287/a72ee0dc-309e-45be-8a37-7273096799d6)

So I guess this is an issue with my model or so. I'll investigate this further.

**Reply by renecotyfanboy** [reply]
Rerunning this in a fresh environment lead to expected results, I probably have an incompatible cocktail package in my dev environment 🥲 

![comparison(2)](https://github.com/Joshuaalbert/jaxns/assets/49200287/1b094502-a6e6-401a-be11-1ddafd485d58)

**Reply by renecotyfanboy** [reply, marked as answer]
After a full day of tracking the why of this issue, it simply comes from the prior distributions I chose, which were too restrictive for `jaxns` to work properly. I moved some to LogUniform instead of Uniform and expanded them a bit and it now works like a charm. So, not a package issue or similar.

**Comment by Joshuaalbert** [comment]
Glad you figured it out!
